SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -866,11 +906,24 @@ void CMeshSolver::Restart_OldGeometry(CGeometry *geometry, CConfig *config) {
 
     if (Unst_RestartIter < 0) {
 
-      if (rank == MASTER_NODE) cout << "Requested mesh restart filename is negative. Setting known solution" << endl;
+      if (rank == MASTER_NODE) cout << "Requested mesh restart filename is negative. Setting zero displacement" << endl;
+
+      unsigned long iPoint_Global;
 
-      /*--- Set loaded solution into correct previous time containers. ---*/
-      if(iStep==1) nodes->Set_Solution_time_n();
-      else nodes->Set_Solution_time_n1();
+      for (iPoint_Global = 0; iPoint_Global < geometry->GetGlobal_nPointDomain(); iPoint_Global++) {
+
+        auto iPoint_Local = geometry->GetGlobal_to_Local_Point(iPoint_Global);
+
+        if (iPoint_Local >= 0) {
+
+          for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+            if(iStep==1)
+              nodes->Set_Solution_time_n(iPoint_Local, iDim, 0.0);
+            else
```

Comment:
To set to zero you can simply loop over nPoint directly.
The `if(iStep...` should be outside of the loops that set to zero.

Reply:
I've updated to loop over nPoint now. The set calls to time_n and time_n1 still depend on which step was being considered, so I've left that the same. If I've misunderstood the change you meant, please let me know or suggest the change which you think is more efficient.

Reply:
Thanks that should be fine.